modules/common/react/lib/utils/getTransformFromPlacement.ts

```diff
@@ -0,0 +1,33 @@
+import {TransformOrigin} from '../types';
+import {Placement} from '@popperjs/core';
+
+export const getTransformFromPlacement = (placement: Placement): TransformOrigin => {
+  const [first, second] = placement.split('-');
+
+  const vertical =
+    first === 'top'
+      ? 'bottom'
+      : first === 'bottom'
+      ? 'top'
+      : second === 'end'
+      ? 'top'
+      : second === 'start'
+      ? 'bottom'
+      : 'center';
+
+  const horizontal =
+    first === 'left'
+      ? 'right'
+      : first === 'right'
+      ? 'left'
+      : second === 'end'
+      ? 'left'
+      : second === 'start'
+      ? 'right'
+      : 'center';
```

Comment:
Not the easiest to read, but well tested. I did TDD for this transformation originally manually setting up all the inputs and outputs hard-coded and then extracted the series of conditionals that lead to the transformation.

This style takes a bit to learn how to read, but it is close to pattern matching. It reads similar to a `switch` statement where if a match is true, it will return the next line, otherwise it will evaluate the next match finally defaulting to the last value.

I don't anticipate any changes to this code. If it takes a bit to understand, I think that's okay. The unit tests are very clear and each input/output is manually extracted.